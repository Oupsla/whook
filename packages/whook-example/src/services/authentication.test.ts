import initAuthentication from './authentication';
import YError from 'yerror';

describe('authentication', () => {
  const TOKEN = 'my_secret';

  describe('.check()', () => {
    describe('with bearer type', () => {
      it('should work with a good token', async () => {
        const authentication = await initAuthentication({ TOKEN });
        const result = await authentication.check('bearer', { hash: TOKEN });

        expect({
          result,
        }).toMatchSnapshot();
      });

      it('should fail with a bad token', async () => {
        const authentication = await initAuthentication({ TOKEN });

        try {
          await authentication.check('bearer', { hash: 'lol' });
          throw new YError('E_UNEXPECTED_SUCCESS');
        } catch (err) {
          expect({
            errorCode: err.code,
            errorParams: err.params,
          }).toMatchSnapshot();
        }
      });
    });

    describe('with fake type', () => {
      it('should work with fakedata', async () => {
        const authentication = await initAuthentication({ TOKEN });
        const result = await authentication.check('fake', {
          userId: 1,
          scopes: ['user'],
        });

        expect({
          result,
        }).toMatchSnapshot();
      });
    });

    describe('with a bad auth type', () => {
      it('should fail', async () => {
        const authentication = await initAuthentication({ TOKEN });

        try {
          await authentication.check('yolo', { hash: 'lol' });
          throw new YError('E_UNEXPECTED_SUCCESS');
        } catch (err) {
          expect({
            errorCode: err.code,
            errorParams: err.params,
          }).toMatchSnapshot();
        }
      });
    });
  });
});
