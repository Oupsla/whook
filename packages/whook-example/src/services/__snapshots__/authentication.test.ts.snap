// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authentication .check() with a bad auth type should fail 1`] = `
Object {
  "errorCode": "E_UNEXPECTED_AUTH_TYPE",
  "errorParams": Array [
    "yolo",
  ],
}
`;

exports[`authentication .check() with bearer type should fail with a bad token 1`] = `
Object {
  "errorCode": "E_BAD_BEARER_TOKEN",
  "errorParams": Array [
    "bearer",
    "lol",
  ],
}
`;

exports[`authentication .check() with bearer type should work with a good token 1`] = `
Object {
  "result": Object {
    "scopes": Array [
      "admin",
    ],
    "userId": 1,
  },
}
`;

exports[`authentication .check() with fake type should work with fakedata 1`] = `
Object {
  "result": Object {
    "scopes": Array [
      "user",
    ],
    "userId": 1,
  },
}
`;
